x-pack/libbeat/common/aws/acker.go

```diff
@@ -18,54 +18,40 @@ import (
 // more S3 objects.
 type EventACKTracker struct {
 	sync.Mutex
-	PendingACKs int64
-	ctx         context.Context
```

Comment:
`ctx` was removed: now when `cancel` is called `Wait` won't return if `ACK` is not called for every `Add` call made.

if we want to keep `ctx` not slow down the shutdown we must deal with https://github.com/elastic/beats/pull/38146/files#r1510803456, so that an error is returned when reading JSON content (and add it to plaintext content as well) so that we can send back the message to the queue and ensure at-least-once delivery